tchannel/peer_heap.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+
+from .container import heap
+from .container.heap import HeapOperation
+
+
+class PeerHeap(HeapOperation):
+    def __init__(self):
+        self.peers = []
+
+    def cmp(self, i, j):
+        return self.peers[i] < self.peers[j]
```

Comment:
Because there's no semantic meaning to a peer being less than another peer. Availability of a feature doesn't mean it's open to misuse. For things like operator overloading, it's very easy to end up with code that is not obvious by looking at it. That's why you should use operator overloading only when there's an actual meaning of the comparison on the object being compared. If you were comparing a vector, there's a specific meaning to "vector A is less than B", so it's okay to define `__lt__` there.